rs/execution_environment/src/query_handler/query_cache.rs

```diff
@@ -151,7 +151,7 @@ impl From<&Query> for EntryKey {
         Self {
             source: match query.source {
                 QuerySource::User { user_id, .. } => user_id,
-                QuerySource::Anonymous => UserId::from(PrincipalId::new_anonymous()),
+                QuerySource::Anonymous => UserId::from(PrincipalId::default()),
```

Comment:
Could we make it explicit that this is the management canister id? Just because it's not so obvious that that's the default.

Reply:
There's a reason that I did not use the management canister id back when I refactored the code and introduced `QuerySource`. The reason was that if the management canister id is used, then we're creating a `UserId` out of a Principal that describes a canister. It technically works but it's weird.

I missed that composite queries have the wrong caller ID so we should fix this but I think we need to come up with a better long term approach that doesn't mix up user and canister ids.